### DISCUSSION TITLE
Query 패키지 내 조회 메소드 관련

### DISCUSSION THREAD

**Discussion by bbbbooo** [goalapa/spring-cacamuca#63] [Ideas]
## @Transactional(readOnly = true) 어노테이션에 대해 어떻게 생각하시나요?

조회 메소드에서 아마 위의 어노테이션을 사용하지 않을거라고 생각합니다. 해당 어노테이션을 사용했을때 얻을 수 있는 이점은 다음과 같습니다.

1. db vendor(oracle, mysql..등등)와 그것의 버전에 따라서 조회만 수행했음에도 불구하고 insert or update가 수행될 수 있습니다. 결국, 해당 어노테이션을 통해 해당 이슈를 방지할 수 있습니다.

2. 데이터베이스의 트렌젝션이 수행될 때, 스냅샷을 저장하지 않고 더티체킹을 수행하지 않아 최적화가 됩니다. 이를 통해서 read의 성능을 향상시킬 수 있습니다.

3. 규모가 작은 프로젝트에서는 데이터베이스가 1개지만 규모가 커지면 master/slave 구조로 나뉘게 됩니다. 이때 readonly=true를 통해 cud는 master db에서 수행하고 r은 slave에서 수행함으로써 부하를 줄일 수 있습니다.

4. 어노테이션을 달아줌으로써, 가독성이 향상됩니다.

따라서, 해당 조회 메소드에서 어노테이션을 사용하는 것이 좋다고 생각하는데 여러분은 어떻게 생각하시나요?

**Comment by numerical43** [comment]
좋은 의견이라고 생각합니다! readOnly에 대해 들어본적은 있지만 어떤 기능인지 깊게 생각해보지않았는데, 이번에 적용해보면 좋을 것같네요👍

**Comment by myrhymetree** [comment]
넵 명시적으로 작성하는 것도 좋을 것 같습니다.
